### DISCUSSION TITLE
Melcloud mode

### DISCUSSION THREAD

**Discussion by Tiwas** [OlivierZal/com.melcloud#50] [General]
Hi,
What's the difference between "set mode" and "set full thermostat mode"? No matter what I set it to, it seems the aircon ends up in "homekit thermostat mode". Is that normal?

I want it set to work off the thermostat, auto pane, auto speed, only heat. Is there an easy way to set that?

**Comment by OlivierZal** [comment]
Hi @Tiwas,

Thanks for your question. Here is some clarification.

`set full thermostat mode` handles every mode of your MELCloud aircon:
- Heat
- Cool
- Auto
- Fan
- Dry

`set mode` ("HomeKit Thermostat Mode") handles only the Homey thermostat built-in modes (which are the ones handled by the Apple Home app – you can plug to it through the HomeyKit app):
- Heat
- Cool
- Auto
- Off

There is an invisible mapping which is done between both:
**From `set mode` to `set full thermostat mode`**
- Heat => Heat
- Cool => Cool
- Auto => Auto
- Off => device is turned "off" (= power button) but mode remains

**From `set full thermostat mode` to `set mode`**
- Heat => Heat
- Cool => Cool
- Auto => Auto
- Fan => Off (but device / power button remains "on")
- Dry => Off (but device / power button remains "on")

If you turn "off" your device (= power button), "Full Thermostat Mode" will remain but "HomeKit Thermostat Mode" will become "Off".

So for heating I advice you to use the `set mode` because it will also handle the "power button" (if your device is off, it will be turned off), although the `set thermostat mode` will just change the mode but won't affect the "power button" (which would need to be done separately).

Does this answer your question? Don't hesitate to provide me precise use cases, and I'll try to assist you in your automation.

Kind regards

**Reply by Tiwas** [reply]
Thanks for replying.

Here's my flow (which probably looks like...):
![image](https://user-images.githubusercontent.com/17053402/201959133-695c4882-428c-45ed-bf1d-9c9feba00b3d.png)

Now, this leaves my aircon in Homekit mode
![image](https://user-images.githubusercontent.com/17053402/201959313-f5aabf47-0c0d-4b32-8c92-b80a2b035690.png)

I've tried manually change it to full thermostat mode, but the next time it runs (which is on the hour) it switches back to homekit.

Any ideas? Also, anything here I can leave out? The gf might change settings using the remote if she doesn't want the wind in her neck (or wants *all* the wind in her neck - depends on the moon phases or something weird) I need to set it back to auto to prevent the rest of my apartment turning into an ice box.

**Comment by OlivierZal** [comment]
That does not seem to be an issue: both HomeKit and Full thermostat modes exists at the same time and are synced one with the other.

When you go to your device type: that's always the first one which is displayed, but that does not mean the second is not well filled.

Anyway: "heat" is the same either from HomeKit or Full thermostat mode, so provided you see heat in your device, that means your workflow worked.

**Reply by Tiwas** [reply]
Ok, thanks. Now I just need to find out why the flow sometimes set the temp to 19C, while running the flow manually sets it to 22C...

**Reply by OlivierZal** [reply]
From what I see, the same contexts can lead to 2 different target temperatures depending on the hour your flow is run (manually or not).
